sql/core/src/main/scala/org/apache/spark/sql/execution/command/ShowCollationsCommand.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.command
+
+import org.apache.spark.sql.{Row, SparkSession}
+import org.apache.spark.sql.catalyst.analysis.ResolvedNamespace
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeReference}
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.util.CollationFactory.Collation
+import org.apache.spark.sql.types.StringType
+
+/**
+ * A command for `SHOW COLLATIONS`.
+ *
+ * The syntax of this command is:
+ * {{{
+ *    SHOW identifier? COLLATIONS ((FROM | IN) ns=identifierReference)? (LIKE? pattern=stringLit);
+ * }}}
+ */
+case class ShowCollationsCommand(
+    ns: LogicalPlan,
+    userScope: Boolean,
+    systemScope: Boolean,
+    pattern: Option[String]) extends UnaryRunnableCommand {
+
+  override val output: Seq[Attribute] = Seq(
+    AttributeReference("COLLATION_CATALOG", StringType, nullable = false)(),
+    AttributeReference("COLLATION_SCHEMA", StringType, nullable = false)(),
+    AttributeReference("COLLATION_NAME", StringType, nullable = false)(),
+    AttributeReference("LANGUAGE", StringType)(),
+    AttributeReference("COUNTRY", StringType)(),
+    AttributeReference("ACCENT_SENSITIVITY", StringType, nullable = false)(),
+    AttributeReference("CASE_SENSITIVITY", StringType, nullable = false)(),
+    AttributeReference("PAD_ATTRIBUTE", StringType, nullable = false)(),
+    AttributeReference("ICU_VERSION", StringType)())
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val ResolvedNamespace(catalog, ns, _) = child
+
+    val systemCollations: Seq[Collation] = if (systemScope) {
+      sparkSession.sessionState.catalog.listCollations(pattern)
+    } else Seq.empty
+
+    val userCollations: Seq[Collation] = if (userScope) {
```

Comment:
This implementation has already left extensions for future third parties.

Reply:
```sql
(base) ➜  spark-community git:(show_collation_syntax) ✗ sh bin/spark-sql
WARNING: Using incubator modules: jdk.incubator.vector
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
Spark Web UI available at http://172.26.2.21:4040
Spark master: local[*], Application Id: local-1725623182529
spark-sql (default)> SHOW user collations;
Time taken: 0.459 seconds
spark-sql (default)> SHOW system collations;
SYSTEM	BUILTIN	UNICODE			ACCENT_SENSITIVE	CASE_SENSITIVE	NO_PAD	75.1.0.0
SYSTEM	BUILTIN	UNICODE_AI			ACCENT_SENSITIVE	CASE_INSENSITIVE	NO_PAD	75.1.0.0
SYSTEM	BUILTIN	UNICODE_CI			ACCENT_INSENSITIVE	CASE_SENSITIVE	NO_PAD	75.1.0.0
SYSTEM	BUILTIN	UNICODE_CI_AI			ACCENT_INSENSITIVE	CASE_INSENSITIVE	NO_PAD	75.1.0.0
SYSTEM	BUILTIN	UTF8_BINARY	NULL	NULL	ACCENT_SENSITIVE	CASE_SENSITIVE	NO_PAD	NULL
SYSTEM	BUILTIN	UTF8_LCASE	NULL	NULL	ACCENT_SENSITIVE	CASE_INSENSITIVE	NO_PAD	NULL
...
```

Reply:
What do you think, @mihailom-db

Reply:
This looks great. Could we maybe force UTF8_BINRAY and UTF8_LCASE to be the first ones? These are spark internal implementations and I would expect the default collation to appear first. What do you think?

Reply:
> This implementation has already left extensions for future third parties.

The future might not come in real life. Let's add this when it is needed.

Reply:
> This looks great. Could we maybe force UTF8_BINRAY and UTF8_LCASE to be the first ones? These are spark internal implementations and I would expect the default collation to appear first. What do you think?

Okay, I completely agree, let me update it.

Reply:
> > This implementation has already left extensions for future third parties.
> 
> The future might not come in real life. Let's add this when it is needed.

Okay, let me update it.